### ISSUE TITLE
vine: zero values in time areas in transactions log

### ISSUE THREAD

**Issue by JinZhou5042** [cooperative-computing-lab/cctools#3864]
I found in the transactions log some `CACHE_UPDATE` records the `wall_time` and `start_time` are `0`, I am guessing there is a bug in the logging function.

<img width="1182" alt="image" src="https://github.com/cooperative-computing-lab/cctools/assets/142265839/a4711418-ad04-461a-a16d-c563170b0967">

**Comment by JinZhou5042** [comment]
I think this issue had been addressed in a more recent update (I extracted those lines from Connor's log file, and he was using an older version). Now I think it's a different problem:

I matched the lines in `debug` file with `cache-update .* X` where 'X' indicates a newly created temp file. It turns out that all newly created temp files have `0` in their size fields (the third field after the file name), for example
```
2024/06/15 15:31:30.33 vine_manager[4141059] vine: rx from d12chas310.crc.nd.edu (10.32.84.127:44562): cache-update temp-rnd-tuswduszjzfjxbm 1 0 0 33188 10269273 1718479837323798 X
2024/06/15 15:31:30.33 vine_manager[4141059] vine: rx from d12chas300.crc.nd.edu (10.32.84.107:36810): cache-update temp-rnd-itjtcdufkwhhjqx 1 0 0 33188 10198761 1718479836672333 X
2024/06/15 15:31:30.33 vine_manager[4141059] vine: rx from d12chas179.crc.nd.edu (10.32.83.121:43724): cache-update temp-rnd-ofdjmychbnvzzjw 1 0 0 33188 10081782 1718479837556866 X
2024/06/15 15:31:30.33 vine_manager[4141059] vine: rx from d12chas117.crc.nd.edu (10.32.82.253:52712): cache-update temp-rnd-hncnylwtftjwubs 1 0 0 33188 9853472 1718479836955804 X
2024/06/15 15:31:30.33 vine_manager[4141059] vine: rx from d12chas185.crc.nd.edu (10.32.83.133:49704): cache-update temp-rnd-ervgivalnekcwyr 1 0 0 33188 9643875 1718479837137254 X
2024/06/15 15:31:30.33 vine_manager[4141059] vine: rx from d12chas163.crc.nd.edu (10.32.83.89:51000): cache-update temp-rnd-pnowkagcgtsumdl 1 0 0 33188 10366510 1718479837350099 X
2024/06/15 15:31:30.33 vine_manager[4141059] vine: rx from d12chas187.crc.nd.edu (10.32.83.137:42544): cache-update temp-rnd-mnjxbnsgryejghe 1 0 0 33188 10021811 1718479836666093 X
2024/06/15 15:31:30.33 vine_manager[4141059] vine: rx from ta-a6k-001.crc.nd.edu (10.32.90.113:42880): cache-update temp-rnd-yrlhecrqgoktvqj 1 0 0 33188 8459983 1718479837472896 X
2024/06/15 15:31:30.33 vine_manager[4141059] vine: rx from d12chas154.crc.nd.edu (10.32.83.71:33758): cache-update temp-rnd-nixetnhkazycovg 1 0 0 33188 14046949 1718479837530555 X
```

I think it makes sense if that is used to report the creation of a temp file to the manager.

**Comment by JinZhou5042** [comment]
But what looks strange to me is that the temp file was only reported once by the worker which actually created it but the reported file size was 0. But then that file was duplicated among multiple workers, each of the cache updates by those workers come with its actual file size.

**Comment by dthain** [comment]
That does seem strange.  The cache-update message comes when the file is moved from the sandbox back to the cache, so the full size should be known at that point.

**Comment by JinZhou5042** [comment]
My hypothesis is that when the worker tries to measure a newly created output file, that file has not been completed closed, therefore it always gets a zero file length.

**Comment by dthain** [comment]
Well, take a look and see!

**Comment by JinZhou5042** [comment]
addressed in #3867
